arrow-libs/core/arrow-core/src/commonTest/kotlin/arrow/core/IterableTest.kt

```diff
@@ -48,6 +48,40 @@ class IterableTest : UnitSpec() {
       }
     }
 
+    "traverseResult stack-safe" {
+      // also verifies result order and execution order (l to r)
+      val acc = mutableListOf<Int>()
+      val res = (0..20_000).traverseResult { a ->
+        acc.add(a)
+        Result.success(a)
+      }
+      res shouldBe Result.success(acc)
+      res shouldBe Result.success((0..20_000).toList())
+    }
+
+    "traverseResult short-circuit" {
+      checkAll(Arb.list(Arb.int())) { ints ->
+        val acc = mutableListOf<Int>()
+        val evens = ints.traverseResult {
+          if (it % 2 == 0) {
+            acc.add(it)
+            Result.success(it)
+          } else Result.failure(RuntimeException())
+        }
+        acc shouldBe ints.takeWhile { it % 2 == 0 }
+        evens.fold(
+          { it shouldBe ints },
+          { }
```

Comment:
Should we add a check on the other side? If we make the exception in L69 ew can then make sure it arrives here :)

Reply:
Actually, this case should be covered. This is the verbose version.

```kotlin
        evens.fold(
          {
             acc shouldBe ints.takeWhile { it % 2 == 0 }
             it shouldBe ints
           },
          { acc shouldBe ints.takeWhile { it % 2 == 0 } }
        )
```